### PR TITLE
feat: add db counter

### PR DESCRIPTION
**Pull request by craigpastro**
## Description

TODO: add dbCounter to the streaming context

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: storage/wrapper.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	Reads  atomic.Int32
+	Writes atomic.Int32
+}
+
+// StorageWrapper is a wrapper around any OpenFGADatastore and is used to record metadata around how the underlying
+// implementation is being used.
+//
+// Currently, we are using it to record the number of reads and writes that are performed. Note that the values for the
+// reads and writes may not be accurate as we are only measuring calls that flow through the interface, and, such a call
+// may involve any number of database accesses under the hood.
```
Comment (miparnisari) [inline storage/wrapper.go:27]: > the values for the  reads and writes may not be accurate as we are only measuring calls that flow through the interface, and, such a call may involve any **number of database accesses under the hood**

but this is exactly what i'd expect to see, especially if we are using one of the iterators... each call to `fetchResults()` should add 1 to the `read` counter

File: storage/wrapper.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	"sync/atomic"
+	"time"
+
+	openfgapb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+)
+
+type ctxKey string
+
+const (
+	DBCounterCtxKey ctxKey = "db_counter"
```
Comment (miparnisari) [inline storage/wrapper.go:14]: This used to be `db_stats`, i wouldn't change it or we'll have to remember when querying old data

Also, do we need db stats at all in OpenFGA?

File: storage/wrapper.go (hunk 1 above)
Comment (craigpastro) [reply storage/wrapper.go:27]: I think that is too much work for too little value. We would have to add it to each implementation. As it is now, it gives a general idea... which there may be some value in:

File: storage/wrapper.go (hunk 2 above)
Comment (craigpastro) [reply storage/wrapper.go:14]: I think the msg key used to be `db_stats`. I think a whole log line dedicated to logging db_stats is a bit overkill and adds too much noise. In any case, I don't think we actually ever queried on it did we? We do have a dashboard widget with it, but, again, I don't think it adds much value.

File: storage/wrapper.go (hunk 1 above)
Comment (craigpastro) [reply storage/wrapper.go:27]: FWIW it definitely did not have that accuracy before either. In fact, it is more accurate now than before.

**Pull request closed without merging**